src/SLCore.IntegrationTests/FileExclusionsTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Reflection;
+
+namespace SonarLint.VisualStudio.SLCore.IntegrationTests;
+
+[TestClass]
+public class FileExclusionsTests
+{
+    private const string someOtherFileExclusion = "**/someotherpath/**";
+    private static FileAnalysisTestsRunner sharedFileAnalysisTestsRunner;
+
+    public TestContext TestContext { get; set; }
+
+    [ClassInitialize]
+    public static async Task ClassInitialize(TestContext context) =>
+        sharedFileAnalysisTestsRunner = await FileAnalysisTestsRunner.CreateInstance(nameof(FileExclusionsTests));
+
+    [ClassCleanup]
+    public static void ClassCleanup() => sharedFileAnalysisTestsRunner.Dispose();
+
+    [DataTestMethod]
+    [DynamicData(nameof(TestingFiles))]
+    public async Task CurrentFileExcluded_CurrentFileNotAnalyzed(ITestingFile testingFile)
+    {
+        var configScope = GetConfigurationScopeName(testingFile);
+        string[] fileExclusions = [someOtherFileExclusion, ("**\\" + testingFile.RelativePath).Replace(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar)];
```

Comment:
We should use the `AnalysisSettings` so that is clear why we use `**\\` and so that we can test backslashes case as well.

```suggestion
      var analysisSettings = new AnalysisSettings()
{
    UserDefinedFileExclusions = [someOtherFileExclusion, testingFile.RelativePath.Replace(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar)]
};
```

Reply:
It feels like a repetition of AnalysisSettingsTests though